src/vs/editor/common/viewModel/splitLinesCollection.ts

```diff
@@ -1327,9 +1323,16 @@ export class SplitLine implements ISplitLine {
 		return (deltaLineNumber + r.outputLineIndex);
 	}
 
-	public leftNormalizePosition(model: ISimpleModel, modelLineNumber: number, outputLineIndex: number, outputPosition: Position): Position {
-		if (outputLineIndex > 0 && outputPosition.column === this._getViewLineMinColumn(outputLineIndex)) {
-			return new Position(outputPosition.lineNumber - 1, this.getViewLineMaxColumn(model, modelLineNumber, outputLineIndex - 1));
+	public normalizePosition(model: ISimpleModel, modelLineNumber: number, outputLineIndex: number, outputPosition: Position, affinity: PositionNormalizationAffinity): Position {
+		if (affinity === PositionNormalizationAffinity.Left) {
+			if (outputLineIndex > 0 && outputPosition.column === this._getViewLineMinColumn(outputLineIndex)) {
+				return new Position(outputPosition.lineNumber - 1, this.getViewLineMaxColumn(model, modelLineNumber, outputLineIndex - 1));
+			}
+		}
+		else if (affinity === PositionNormalizationAffinity.Right) {
+			if (outputPosition.column === this.getViewLineMaxColumn(model, modelLineNumber, outputLineIndex)) {
```

Comment:
I think there might be a bug here because this if statement will go through even on the last consecutive wrapped line. In this case, the if statement should not go through:
<img width="266" alt="Screen Shot 2021-05-10 at 12 01 08" src="https://user-images.githubusercontent.com/5047891/117642341-749bc900-b187-11eb-9897-06d4a2f797ff.png">